packages/synth.queues/src/services/kafka.queue.service.ts

```diff
@@ -0,0 +1,118 @@
+import { PluginManager } from '@applicature/synth.plugin-manager';
+import * as kafka from 'node-rdkafka';
+import * as logger from 'winston';
+import { KafkaConfig, Message, Queue } from '../types';
+import { QueueService } from './queue.service';
+export class KafkaQueueService extends QueueService {
+    private producerReady: any;
+    private consumerReady: any;
+    private adminClient: kafka.InternalAdminClient;
+    private topics: Set<string>;
+
+    // tslint:disable-next-line:no-shadowed-variable
+    constructor(pluginManager: PluginManager, config: KafkaConfig) {
+        super(pluginManager);
+
+        const producer = new kafka.Producer(config.producerConfig, config.topicConfig);
+        producer.connect(
+            {},
+            (err: any) => {
+                if (err) {
+                    logger.error('connect', err);
+                }
+            }
+        );
+        this.producerReady = new Promise((resolve: any) => {
+            producer.on('ready', () => {
+                logger.info('producer ready');
+                resolve(producer);
+            });
+        });
+
+        const consumer = new kafka.KafkaConsumer(config.consumerConfig, config.topicConfig);
+        consumer.connect(
+            {},
+            (err: any) => {
+                if (err) {
+                    logger.error('connect', err);
+                }
+            }
+        );
+        this.consumerReady = new Promise((resolve: any) => {
+            consumer.on('ready', () => {
+                logger.info('consumer ready');
+                resolve(consumer);
+            });
+        });
+
+        const client = new kafka.AdminClient();
+        this.adminClient = client.create(config.adminConfig);
+    }
+
+    public async createQueue(
+        queueName: string,
+        numPartitions: number,
+        replicationFactor: number,
+        topicConfig: object
+    ): Promise<Queue> {
```

Comment:
create queue should contain 1 parameter: `createQueue(queueName: string): Promsie<Queue>`